pm: Adds some basic pm stats

teburd opened the pull request:
Adds stats to the system power management state changes in place of the specific pm debug stats that were previously tracked.

MaureenHelm reviewed (commented):
I had to stare at the second commit for a while to understand what it's doing. The commit message is a little sparse and could benefit from some elaboration.

teburd commented:
I'm moving the shell command bits to a separate PR here: https://github.com/zephyrproject-rtos/zephyr/pull/39345 and have incorporated the changes on those files

ceolin reviewed (approved):
There are references to `CONFIG_PM_DEBUG` that need to be fixed. Other than that, looks ok for me.

Commit:
pm: Use stats subsys for tracking system states

Uses the stats subsys to provide simple but useful debugging stats for
power management state changes and timing.

Removes the no longer needed PM_DEBUG config option

Replaces the use of PM_DEBUG for a test clock output pin for mec1501 and
adds in its place an SoC Kconfig option to enable it.

Adds a STATS_SET macro for assigning a value to a stat group field

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>
